### DISCUSSION TITLE
Socket : how to find the indexes following a recv

### DISCUSSION THREAD

**Discussion by PatLub85** [micropython/micropython#12362] [General]
Hi,
I have an ESP32 server which offers the lighting of an LED via an HTML page.
All the examples I've seen on the internet show this line:
if xxx == 6:
    do this.
I saw only one explanation on this site (https://microcontrollerslab.com/esp32-esp8266-micropython-web-server/)but I find it too succinct.
The substrings '/?led_2_on' or '/?led_2_off' are always on the 6th index !!!
![image](https://github.com/micropython/micropython/assets/60620342/2017cd4d-b0dc-4b3b-90ef-cc9e4e23afb5)

```s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
s.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
s.bind((ip, 80))
s.listen(5)
while True:
    conn, addr = s.accept()
    print('Got a connection from %s' % str(addr))
    request = conn.recv(1024)
    request = str(request)
    print('Content = %s' % request)
    led_on = request.find('/?led_2_on')
    led_off = request.find('/?led_2_off')
    if led_on == 6:
        print('LED ON -> GPIO2')
        led_state = "ON"
        led.on()
```
My question is : how to find and print the other indexes

**Comment by shariltumin** [comment]
You can give this a try:

```py
s.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
s.bind((ip, 80))
s.listen(5)
while True:
    conn, addr = s.accept()
    print('Got a connection from %s' % str(addr))
    request = conn.recv(1024)
    print('Content = %s' % request.decode())
    if b'/?led_2_on' in request:
       print('LED ON -> GPIO2')
       led_state = "ON"
       led.on()
    elif b'/?led_2_off' in request:
       print('LED OFF -> GPIO2')
       led_state = "OFF"
       led.off()
    else:
       print('Unknown CMD')

```

**Reply by PatLub85** [reply]
Thanks Chariltumine
your solution is the most readable and logical. I'm keeping it for my project.
On the other hand, as I like to understand what I'm doing, I still don't know what index 5, 7, etc. represents, for example.
Does anyone have the explanation?

**Reply by bixb922** [reply]
request.find returns the position where the byte string ```/?led_2_off``` occurs in the request bytearray. The first position is zero, so a value of 6 means there are 6 bytes of data in the request before the start of the substring '/?led_2_off`
Is that what you are looking for?
